src/bluesearch/database/topic.py

```diff
@@ -0,0 +1,229 @@
+    except ElementTree.ParseError:
+        # Occurs when the number of results of the research is bigger than one.
+        # It is the case for less than 1 % of the journal from PMC
+        raise ElementTree.ParseError("The parsing did not work")
+
+    mesh_headings = content.findall("./NLMCatalogRecord/MeshHeadingList/MeshHeading")
+    meshs = get_mesh_from_nlm_catalog(mesh_headings)
+
+    return meshs
+
+
+# Article Topic
+def get_mesh_from_pubmed_id(pubmed_ids: Iterable[str]) -> Dict:
+    """Retrieve Medical Subject Headings from Pubmed ID.
+
+    Parameters
+    ----------
+    pubmed_ids
+        List of Pubmed IDs.
+
+    Returns
+    -------
+    pubmed_to_meshs : dict
+        Dictionary containing Pubmed IDs as keys with corresponding
+        Medical Subject Headings list as values.
+    """
+    pubmed_str = ",".join(pubmed_ids)
+    url = (
+        "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?"
+        f"db=pubmed&id={pubmed_str}&retmode=xml"
+    )
+    response = requests.get(url)
+
+    if not response.ok:
+        raise ValueError("The request did not work")
+
+    content = ElementTree.fromstring(response.content.decode())
+    pubmed_articles = content.findall("./PubmedArticle")
+    pubmed_to_meshs = {}
+
+    for article in pubmed_articles:
+        pubmed_id = article.findall(
+            "./PubmedData/ArticleIdList/ArticleId[@IdType='pubmed']"
+        )[0].text
+        mesh_headings = article.findall("./MedlineCitation/MeshHeadingList")
+        meshs = get_mesh_from_pubmed(mesh_headings)
+        pubmed_to_meshs[pubmed_id] = meshs
+
+    return pubmed_to_meshs
+
+
+# Utils
+def get_pubmed_id_from_pmc_file(path: Union[str, pathlib.Path]) -> Optional[str]:
+    """Retrieve Pubmed ID from PMC XML file.
+
+    Parameters
+    ----------
+    path
+        Path to PMC XML.
+
+    Returns
+    -------
+    pubmed_id : str
+        Pubmed ID of the given article
+    """
+    content = ElementTree.parse(path)
+    article_id = content.findall("./front/article-meta/article-id[@pub-id-type='pmid']")
+
+    if len(article_id) == 1:
+        pubmed_id = article_id[0].text
+    else:
+        pubmed_id = None
+
+    return pubmed_id
+
+
+def get_mesh_from_nlm_catalog(mesh_headings: Iterable[Element]) -> List[Dict]:
+    """Retrieve Medical Subject Headings from nlmcatalog parsing.
+
+    Parameters
+    ----------
+    mesh_headings
+        XML parsing element containing all Medical Subject Headings.
+
+    Returns
+    -------
+    mesh : list of dict
+        List of dictionary containing Medical Subject Headings information.
+    """
+    meshs = []
+    for mesh in mesh_headings:
+
+        mesh_id = None
+        if "URI" in mesh.attrib.keys():
+            mesh_id = mesh.attrib["URI"].rpartition("/")[-1]
+
+        descriptor_name = []
+        qualifier_name = []
+
+        for elem in mesh:
+            major_topic = True if elem.attrib["MajorTopicYN"] == "Y" else False
```

Comment:
I think this can be written as `major_topic = elem.get("MajorTopicYN") == "Y"`

Reply:
Done in 8c32705